apps/application-system/api/src/app/modules/payment/payment.service.ts

```diff
@@ -135,4 +135,28 @@ export class PaymentService {
     }
     return application[0] as ApplicationModel
   }
+
+  async delete(applicationId: string, @CurrentUser() user: User) {
```

Comment:
You don't use the decorator in a service method, as you are always sending the user in as a parameter. 
The decorator is used in a top level  controller method, to read the user from the context.
```suggestion
  async delete(applicationId: string, user: User) {
```